server/service/src/stock_line/update.rs

```diff
@@ -46,7 +48,7 @@ pub fn update_stock_line(
         .connection
         .transaction_sync(|connection| {
             let existing = validate(connection, &ctx.store_id, &input)?;
-            let (new_stock_line, location_movements) =
+            let (new_stock_line, location_movements, barcode_row) =
                 generate(ctx.store_id.clone(), connection, existing.clone(), input)?;
             StockLineRowRepository::new(&connection).upsert_one(&new_stock_line)?;
```

Comment:
@mark-prins 

> To test, I've:
>  Updated stock lines via the view stock modal

I am really surprised this worked, unless you've changed the code since you've done those tests ?

> StockLineRowRepository::new(&connection).upsert_one(&new_stock_line)?;

This line location (before barcode update) results in foreign key constraint violation

Reply:
yeah. perhaps tested before the latest refactor. fixed now